### ISSUE TITLE
Trying to connect my Linux 22.04 Samba Domain to Zentyal getting Error Samba 

### ISSUE THREAD

**Issue by Baltimorepc** [zentyal/zentyal#2155]
Trying to Connect my Linux 22.04 Samba Domain  using Zentyal as an additional Domain Controller and this is the last of the log report ending. if you need me to passed more i will need to make some changes before posting 

Changes made to log :Main Samba AD Domain1 (Linux 22.04 Samba AD. Samba version 4.15.13-Ubuntu

2024/03/14 00:39:53 INFO> Provision.pm:1289 EBox::Samba::Provision::provisionADC - Joining to domain 'domain1.local' as DC
2024/03/14 00:39:53 ERROR> Sudo.pm:240 EBox::Sudo::_rootError - root command set -e
cp /etc/resolvconf/interface-order /var/lib/zentyal/tmp/resolvbMnuL2
echo zentyal.temp > /etc/resolvconf/interface-order
echo 'search Domain1.local
nameserver domain1 ip address' | resolvconf -a zentyal.temp failed.
Error output: cp: cannot stat '/etc/resolvconf/interface-order': No such file or directory

Command output: .
Exit value: 1 at root command set -e
cp /etc/resolvconf/interface-order /var/lib/zentyal/tmp/resolvbMnuL2
echo zentyal.temp > /etc/resolvconf/interface-order
echo 'search domain1.local
nameserver domain1 ip address' | resolvconf -a zentyal.temp failed.
Error output: cp: cannot stat '/etc/resolvconf/interface-order': No such file or directory

**Comment by Baltimorepc** [comment]
![image](https://github.com/zentyal/zentyal/assets/141858534/e6d9ab92-32d4-44d3-9b29-59a6099a7bf8)

**Comment by Baltimorepc** [comment]
when i try to connect Zentyal to main Samba AD

**Comment by theroch** [comment]
Was your version of Ubuntu updated from 20.04 or is this a fresh installation?
Is there a folder /etc/resolveconf present or doesn't exist this folder?

**Comment by Baltimorepc** [comment]
It was a fresh install vm of 22.04 on a esxi 8.0 and then ran the install instructions from zentyal 
And yes it is present 



Sent from Yahoo Mail for iPhone


On Saturday, March 23, 2024, 6:55 AM, Frank Rochlitzer ***@***.***> wrote:




Was your version of Ubuntu updated from 20.04 or is this a fresh installation?
Is there a folder /etc/resolveconf present or doesn't exist this folder?

**Comment by Baltimorepc** [comment]
my main samba AD is running on 

PRETTY_NAME="Ubuntu 22.04.4 LTS"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04.4 LTS (Jammy Jellyfish)"
VERSION_CODENAME=jammy
ID=ubuntu
ID_LIKE=debian
UBUNTU_CODENAME=jammy

**Comment by Baltimorepc** [comment]
i verified both have the same versions of linux and samba.

**Comment by theroch** [comment]
Try to create the file `/etc/resolveconf/interface-order` manually and try again
https://manpages.ubuntu.com/manpages/trusty/man5/interface-order.5.html

**Comment by Baltimorepc** [comment]
im not the best with linux but ill see if i can google it

**Comment by Baltimorepc** [comment]
Do i need to just create the folder?

**Comment by Baltimorepc** [comment]
Thank you I will give that a try today after work 


Sent from Yahoo Mail for iPhone


On Monday, April 15, 2024, 4:33 AM, Frank Rochlitzer ***@***.***> wrote:




Execute the following commands as root or with sudo:
mkdir /etc/resolveconf
touch /etc/resolveconf/interface-order

Edit /etc/resolveconf/interface-order with yout favorite editor, and add lines like this:
# /etc/resolvconf/interface-order
# Use nameservers on the loopback interface first.
lo*
# Next use records for Ethernet interfaces
eth*
# Next use records for Wi-Fi interfaces
wlan*
# Next use records for PPP interfaces
ppp*
# Last use other interfaces
*

**Comment by Baltimorepc** [comment]
Weird thing is i was verifying Zentyal /etc/resolveconf  and its wrong i see /etc/resolvconf 

is there supposed to be an e or no e 

i just verified my main server doesnt have this at all.  So am i supposed to add /etc/resolveconf or /etc/resolvconf

**Comment by Baltimorepc** [comment]
And should i be fixing the zentyal  from /etc/resolvconf to /etc/resolveconf or leave it alone

**Comment by theroch** [comment]
Sorry, my mistake. It is /etc/resolvconf without "e"

**Comment by Baltimorepc** [comment]
Thank you I will try it when I get off tonight o appreciate everyone help and assistance most appreciated 


Sent from Yahoo Mail for iPhone


On Tuesday, April 16, 2024, 10:17 AM, Frank Rochlitzer ***@***.***> wrote:




Sorry, my mistake. It is /etc/resolvconf without "e"

**Comment by Baltimorepc** [comment]
still getting error messages  "changed domain and ip" for security reasons

2024/04/16 19:36:41 INFO> Provision.pm:1062 EBox::Samba::Provision::checkADNebiosName - Checking domain netbios name...
2024/04/16 19:36:42 INFO> Provision.pm:1289 EBox::Samba::Provision::provisionADC - Joining to domain 'mydomain.local' as DC
2024/04/16 19:36:42 ERROR> Sudo.pm:240 EBox::Sudo::_rootError - root command set -e
cp /etc/resolvconf/interface-order /var/lib/zentyal/tmp/resolv4FPedM
echo zentyal.temp > /etc/resolvconf/interface-order
echo 'search mydomain.local
nameserver 10.0.0.1' | resolvconf -a zentyal.temp failed.
Error output: /var/lib/zentyal/tmp/k0WIM7B2PU.cmd: 5: resolvconf: not found

Command output: .
Exit value: 127 at root command set -e
cp /etc/resolvconf/interface-order /var/lib/zentyal/tmp/resolv4FPedM
echo zentyal.temp > /etc/resolvconf/interface-order
echo 'search mydomain.local
nameserver 10.0.0.1' | resolvconf -a zentyal.temp failed.
Error output: /var/lib/zentyal/tmp/k0WIM7B2PU.cmd: 5: resolvconf: not found

Command output: .
Exit value: 127 at /usr/share/perl5/EBox/Sudo.pm line 240
EBox::Sudo::_rootError('/usr/bin/sudo -p sudo: /var/lib/zentyal/tmp/k0WIM7B2PU.cmd 2> /var/lib/zentyal/tmp/stderr', 'set -e^Jcp /etc/resolvconf/interface-order /var/lib/zentyal/tmp/resolv4FPedM^Jecho zentyal.te>EBox::Sudo::_root(1, 'cp /etc/resolvconf/interface-order /var/lib/zentyal/tmp/resolv4FPedM', 'echo zentyal.temp > /etc/resolvconf/interface-order', 'echo \'search mydomain.local^Jnameserver 10.0.0.1\>EBox::Sudo::root('cp /etc/resolvconf/interface-order /var/lib/zentyal/tmp/resolv4FPedM', 'echo zentyal.temp > /etc/resolvconf/interface-order', 'echo \'search mydomain.local^Jnameserver 10.0.0.1\' | >eval {...} at /usr/share/perl5/EBox/Samba/Provision.pm line 1288
EBox::Samba::Provision::provisionADC('EBox::Samba::Provision=HASH(0x564c6cc40b60)') called at /usr/share/perl5/EBox/Samba/Provision.pm line 369
EBox::Samba::Provision::provision('EBox::Samba::Provision=HASH(0x564c6cc40b60)') called at /usr/share/perl5/EBox/Samba.pm line 695
EBox::Samba::_setConf('EBox::Samba=HASH(0x564c6b7d42b0)') called at /usr/share/perl5/EBox/Module/Base.pm line 995
EBox::Module::Base::_regenConfig('EBox::Samba=HASH(0x564c6b7d42b0)') called at /usr/share/perl5/EBox/Module/Service.pm line 940
EBox::Module::Service::_regenConfig('EBox::Samba=HASH(0x564c6b7d42b0)') called at /usr/share/perl5/EBox/Samba.pm line 668
EBox::Samba::_regenConfig('EBox::Samba=HASH(0x564c6b7d42b0)') called at /usr/share/perl5/EBox/Module/Base.pm line 234
eval {...} at /usr/share/perl5/EBox/Module/Base.pm line 233
EBox::Module::Base::save('EBox::Samba=HASH(0x564c6b7d42b0)') called at /usr/share/perl5/EBox/GlobalImpl.pm line 649
eval {...} at /usr/share/perl5/EBox/GlobalImpl.pm line 648
EBox::GlobalImpl::saveAllModules('EBox::GlobalImpl=HASH(0x564c6ab489e0)', 'progress', 'EBox::ProgressIndicator=HASH(0x564c6aadad18)') called at /usr/share/perl5/EBox/Global.pm line 95
EBox::Global::AUTOLOAD('EBox::Global=HASH(0x564c66867330)', 'progress', 'EBox::ProgressIndicator=HASH(0x564c6aadad18)') called at /usr/share/zentyal/global-action line 32
eval {...} at /usr/share/zentyal/global-action line 30
2024/04/16 19:36:42 INFO> Provision.pm:299 EBox::Samba::Provision::setupKerberos - Setting up kerberos
2024/04/16 19:36:42 INFO> Provision.pm:276 EBox::Samba::Provision::setupDNS - Setting up DNS
2024/04/16 19:36:42 INFO> Base.pm:231 EBox::Module::Base::save - Restarting service for module: dns
2024/04/16 19:36:44 INFO> DNS.pm:92 EBox::DNS::appArmorProfiles - Setting DNS apparmor profile
2024/04/16 19:36:47 ERROR> Sudo.pm:240 EBox::Sudo::_rootError - root command set -e
cp /var/lib/zentyal/tmp/resolv4FPedM /etc/resolvconf/interface-order
resolvconf -d zentyal.temp failed.
Error output: /var/lib/zentyal/tmp/gr_x53P0_Q.cmd: 3: resolvconf: not found

**Comment by Baltimorepc** [comment]
Thank you for your time and all the troubleshooting, Cant wait to be able to update and finally use this.

**Comment by Baltimorepc** [comment]
please let me know when the update for this comes out as i just tried the newest update and still same error

**Comment by aleximper** [comment]
Hi,

To solve the error install the openresolv package.
apt install openresolv

**Comment by Baltimorepc** [comment]
to what my main samba domain or Zentyal

**Comment by aleximper** [comment]
Zentyal aditional domain controller

**Comment by Baltimorepc** [comment]
now getting this error

Some modules reported error when saving changes . More information on the logs in /var/log/zentyal/

The following modules failed while saving their changes, their state is unknown: samba dns

**Comment by aleximper** [comment]
Please reboot and retry, put log here

**Comment by Baltimorepc** [comment]
i did reboot after i did it and thats when i looked but i can do it again

**Comment by Baltimorepc** [comment]
most recent log below 

  GNU nano 6.2                                                                                /var/log/zentyal/zentyal.log                                                                                         2024/04/29 15:16:02 INFO> SyncDaemon.pm:131 EBox::Samba::SyncDaemon::checkUsers - Set user 'CN=Angelina Adams,CN=Users,DC=custombuiltpcs,DC=local' gidNumber=2513
2024/04/29 15:16:03 INFO> SyncDaemon.pm:125 EBox::Samba::SyncDaemon::checkUsers - Set user 'CN=nextcloud,CN=Users,DC=custombuiltpcs,DC=local' uidNumber=65539
2024/04/29 15:16:03 INFO> SyncDaemon.pm:131 EBox::Samba::SyncDaemon::checkUsers - Set user 'CN=nextcloud,CN=Users,DC=custombuiltpcs,DC=local' gidNumber=2513
2024/04/29 15:53:17 INFO> Halt.pm:150 EBox::SysInfo::Model::Halt::_prepareSystemForHalt - Zentyal is going down for reboot
2024/04/29 15:53:21 INFO> Init.pm:101 EBox::Util::Init::stop - Modules to stop: webadmin network firewall audit dns logs ntp samba
2024/04/29 15:53:28 INFO> Service.pm:965 EBox::Module::Service::restartService - Restarting service for module: firewall
2024/04/29 15:55:39 INFO> Init.pm:85 EBox::Util::Init::start - Modules to start: network firewall audit dns logs ntp samba webadmin
2024/04/29 15:55:39 INFO> Service.pm:965 EBox::Module::Service::restartService - Restarting service for module: network
2024/04/29 15:55:52 INFO> Service.pm:965 EBox::Module::Service::restartService - Restarting service for module: firewall
2024/04/29 15:55:55 INFO> Service.pm:965 EBox::Module::Service::restartService - Restarting service for module: firewall
2024/04/29 15:55:55 INFO> Service.pm:965 EBox::Module::Service::restartService - Restarting service for module: audit
2024/04/29 15:55:55 INFO> Service.pm:965 EBox::Module::Service::restartService - Restarting service for module: dns
2024/04/29 15:55:59 INFO> DNS.pm:92 EBox::DNS::appArmorProfiles - Setting DNS apparmor profile
2024/04/29 15:56:03 INFO> Service.pm:965 EBox::Module::Service::restartService - Restarting service for module: logs
2024/04/29 15:56:04 INFO> Service.pm:965 EBox::Module::Service::restartService - Restarting service for module: ntp
2024/04/29 15:56:15 INFO> NTP.pm:60 EBox::NTP::appArmorProfiles - Setting NTP apparmor profile
2024/04/29 15:56:19 INFO> Service.pm:965 EBox::Module::Service::restartService - Restarting service for module: samba
2024/04/29 15:56:28 INFO> SyncDaemon.pm:340 EBox::Samba::SyncDaemon::run - Samba sync daemon started
2024/04/29 15:56:28 INFO> LDAP.pm:246 EBox::Module::LDAP::_sendSchemaUpdate - Sending schema update: CN=quota,CN=Schema,CN=Configuration,DC=custombuiltpcs,DC=local
2024/04/29 15:56:28 ERROR> LDAP.pm:248 EBox::Module::LDAP::_sendSchemaUpdate - Error sending schema update: CN=quota,CN=Schema,CN=Configuration,DC=custombuiltpcs,DC=local The server is unwilling to perform the > at Error sending schema update: CN=quota,CN=Schema,CN=Configuration,DC=custombuiltpcs,DC=local The server is unwilling to perform the requested operation
 at /usr/share/perl5/EBox/Module/LDAP.pm line 248
EBox::Module::LDAP::_sendSchemaUpdate('EBox::Samba=HASH(0x562391eeb9c0)', 'Net::LDAP=HASH(0x56239243ec48)', '/usr/share/zentyal-samba/schema-quota.ldif') called at /usr/share/perl5/EBox/Module/LDAP.pm line 278
EBox::Module::LDAP::_loadSchemasFiles('EBox::Samba=HASH(0x562391eeb9c0)', 'ARRAY(0x5623919163a8)') called at /usr/share/perl5/EBox/Module/LDAP.pm line 267
EBox::Module::LDAP::_loadSchemas('EBox::Samba=HASH(0x562391eeb9c0)') called at /usr/share/perl5/EBox/Module/LDAP.pm line 341
EBox::Module::LDAP::_performSetup('EBox::Samba=HASH(0x562391eeb9c0)') called at /usr/share/perl5/EBox/Samba.pm line 669
EBox::Samba::_regenConfig('EBox::Samba=HASH(0x562391eeb9c0)', 'restart', 1, 'restartModules', 1) called at /usr/share/perl5/EBox/Module/Service.pm line 967
eval {...} at /usr/share/perl5/EBox/Module/Service.pm line 966
EBox::Module::Service::restartService('EBox::Samba=HASH(0x562391eeb9c0)', 'restartModules', 1) called at /usr/share/perl5/EBox/Util/Init.pm line 121
eval {...} at /usr/share/perl5/EBox/Util/Init.pm line 119
EBox::Util::Init::moduleAction('samba', 'restartService', 'start') called at /usr/share/perl5/EBox/Util/Init.pm line 87
EBox::Util::Init::start at /usr/bin/zs line 35
main::main at /usr/bin/zs line 82
2024/04/29 15:56:28 ERROR> Service.pm:969 EBox::Module::Service::restartService - Error restarting service: Error sending schema update: CN=quota,CN=Schema,CN=Configuration,DC=custombuiltpcs,DC=local The server>2024/04/29 15:56:28 ERROR> Service.pm:971 EBox::Module::Service::restartService - Error sending schema update: CN=quota,CN=Schema,CN=Configuration,DC=custombuiltpcs,DC=local The server is unwilling to perform t> at Error sending schema update: CN=quota,CN=Schema,CN=Configuration,DC=custombuiltpcs,DC=local The server is unwilling to perform the requested operation
 at /usr/share/perl5/EBox/Module/Service.pm line 971
EBox::Module::Service::restartService('EBox::Samba=HASH(0x562391eeb9c0)', 'restartModules', 1) called at /usr/share/perl5/EBox/Util/Init.pm line 121
eval {...} at /usr/share/perl5/EBox/Util/Init.pm line 119
EBox::Util::Init::moduleAction('samba', 'restartService', 'start') called at /usr/share/perl5/EBox/Util/Init.pm line 87
EBox::Util::Init::start at /usr/bin/zs line 35
main::main at /usr/bin/zs line 82
2024/04/29 15:56:28 INFO> Service.pm:965 EBox::Module::Service::restartService - Restarting service for module: webadmin
2024/04/29 15:56:34 INFO> Init.pm:90 EBox::Util::Init::start - Start modules finished
2024/04/29 15:56:39 INFO> Index.pm:187 EBox::Dashboard::CGI::Index::masonParameters - dashboard1

**Comment by Baltimorepc** [comment]
there is alot in the log i am trying to clear it out so it can be fresh please let me know if its missing something you need 
hopefully i changed what i needed to 
change ether way tis internal only

**Comment by aleximper** [comment]
With the command tail -f /var/log/zentyal/Zentyal.log you will be able to see the live log output while making the configuration to have the exact logs.

**Comment by Baltimorepc** [comment]
here is a re rrun i unchecked the box for domain controller saved and got a error but still removed the uncheck then i rebooted and then tried to rejoin as a additional domain controller again and got another samba error. here is the log for both. i have not removed anything as this is internal but i will delete this log in a couple days for security anyway

Command output: .
Exit value: 1 at /usr/share/perl5/EBox/Sudo.pm line 240
EBox::Sudo::_rootError('/usr/bin/sudo -p sudo: /var/lib/zentyal/tmp/AdoEfNegUw.cmd 2> /var/lib/zentyal/tmp/stderr', 'kinit -k -t /var/lib/samba/private/dns.keytab dns-ADM2', 256, 'ARRAY(0x55637ac0c220)', 'ARRAY>EBox::Sudo::_root(1, 'kinit -k -t /var/lib/samba/private/dns.keytab dns-ADM2') called at /usr/share/perl5/EBox/Sudo.pm line 153
EBox::Sudo::root('kinit -k -t /var/lib/samba/private/dns.keytab dns-ADM2') called at /usr/share/perl5/EBox/DNS.pm line 1005
EBox::DNS::_postServiceHook('EBox::DNS=HASH(0x5563795b3ec0)', 1) called at /usr/share/perl5/EBox/Module/Service.pm line 948
EBox::Module::Service::_regenConfig('EBox::DNS=HASH(0x5563795b3ec0)') called at /usr/share/perl5/EBox/Module/Base.pm line 234
eval {...} at /usr/share/perl5/EBox/Module/Base.pm line 233
EBox::Module::Base::save('EBox::DNS=HASH(0x5563795b3ec0)') called at /usr/share/perl5/EBox/GlobalImpl.pm line 649
eval {...} at /usr/share/perl5/EBox/GlobalImpl.pm line 648
EBox::GlobalImpl::saveAllModules('EBox::GlobalImpl=HASH(0x556378a58e00)', 'progress', 'EBox::ProgressIndicator=HASH(0x5563747a2e00)') called at /usr/share/perl5/EBox/Global.pm line 95
EBox::Global::AUTOLOAD('EBox::Global=HASH(0x556374774470)', 'progress', 'EBox::ProgressIndicator=HASH(0x5563747a2e00)') called at /usr/share/zentyal/global-action line 32
eval {...} at /usr/share/zentyal/global-action line 30
2024/04/29 16:47:45 ERROR> GlobalImpl.pm:653 EBox::GlobalImpl::saveAllModules - Failed to save changes in module dns: root command kinit -k -t /var/lib/samba/private/dns.keytab dns-ADM2 failed.
Error output: kinit: krb5_get_init_creds: unable to reach any KDC in realm CUSTOMBUILTPCS.LOCAL

Command output: .
Exit value: 1
2024/04/29 16:47:45 INFO> Base.pm:231 EBox::Module::Base::save - Restarting service for module: samba
2024/04/29 16:47:49 INFO> Base.pm:231 EBox::Module::Base::save - Restarting service for module: logs
2024/04/29 16:47:49 ERROR> GlobalImpl.pm:728 EBox::GlobalImpl::saveAllModules - The following modules failed while saving their changes, their state is unknown: dns at The following modules failed while saving>EBox::GlobalImpl::saveAllModules('EBox::GlobalImpl=HASH(0x556378a58e00)', 'progress', 'EBox::ProgressIndicator=HASH(0x5563747a2e00)') called at /usr/share/perl5/EBox/Global.pm line 95
EBox::Global::AUTOLOAD('EBox::Global=HASH(0x556374774470)', 'progress', 'EBox::ProgressIndicator=HASH(0x5563747a2e00)') called at /usr/share/zentyal/global-action line 32
eval {...} at /usr/share/zentyal/global-action line 30
2024/04/29 16:49:00 INFO> Halt.pm:150 EBox::SysInfo::Model::Halt::_prepareSystemForHalt - Zentyal is going down for reboot
2024/04/29 16:49:04 INFO> Init.pm:101 EBox::Util::Init::stop - Modules to stop: webadmin network firewall audit dns logs ntp samba
2024/04/29 16:49:09 INFO> Service.pm:965 EBox::Module::Service::restartService - Restarting service for module: firewall
2024/04/29 16:51:22 INFO> Init.pm:85 EBox::Util::Init::start - Modules to start: network firewall audit dns logs ntp samba webadmin
2024/04/29 16:51:22 INFO> Service.pm:965 EBox::Module::Service::restartService - Restarting service for module: network
2024/04/29 16:51:36 INFO> Service.pm:965 EBox::Module::Service::restartService - Restarting service for module: firewall
2024/04/29 16:51:38 INFO> Service.pm:965 EBox::Module::Service::restartService - Restarting service for module: firewall
2024/04/29 16:51:38 INFO> Service.pm:965 EBox::Module::Service::restartService - Restarting service for module: audit
2024/04/29 16:51:38 INFO> Service.pm:965 EBox::Module::Service::restartService - Restarting service for module: dns
2024/04/29 16:51:40 INFO> DNS.pm:92 EBox::DNS::appArmorProfiles - Setting DNS apparmor profile
2024/04/29 16:51:43 INFO> Service.pm:965 EBox::Module::Service::restartService - Restarting service for module: logs
2024/04/29 16:51:44 INFO> Service.pm:965 EBox::Module::Service::restartService - Restarting service for module: ntp
2024/04/29 16:51:55 INFO> NTP.pm:60 EBox::NTP::appArmorProfiles - Setting NTP apparmor profile
2024/04/29 16:51:58 INFO> Service.pm:965 EBox::Module::Service::restartService - Restarting service for module: samba
2024/04/29 16:52:02 INFO> Service.pm:965 EBox::Module::Service::restartService - Restarting service for module: webadmin
2024/04/29 16:52:08 INFO> Init.pm:90 EBox::Util::Init::start - Start modules finished
2024/04/29 16:56:43 INFO> GlobalImpl.pm:617 EBox::GlobalImpl::saveAllModules - Saving config and restarting services: firewall dns samba logs
2024/04/29 16:56:43 INFO> Base.pm:231 EBox::Module::Base::save - Restarting service for module: firewall
2024/04/29 16:56:44 INFO> Base.pm:231 EBox::Module::Base::save - Restarting service for module: dns
2024/04/29 16:56:54 INFO> DNS.pm:92 EBox::DNS::appArmorProfiles - Setting DNS apparmor profile
2024/04/29 16:56:58 INFO> Base.pm:231 EBox::Module::Base::save - Restarting service for module: samba
2024/04/29 16:57:05 INFO> Samba.pm:774 EBox::Samba::_postServiceHook - Writing DNS update list...
2024/04/29 16:57:07 INFO> SyncDaemon.pm:340 EBox::Samba::SyncDaemon::run - Samba sync daemon started
2024/04/29 16:57:08 INFO> LDAP.pm:246 EBox::Module::LDAP::_sendSchemaUpdate - Sending schema update: CN=quota,CN=Schema,CN=Configuration,DC=custombuiltpcs,DC=local
2024/04/29 16:57:08 ERROR> LDAP.pm:248 EBox::Module::LDAP::_sendSchemaUpdate - Error sending schema update: CN=quota,CN=Schema,CN=Configuration,DC=custombuiltpcs,DC=local The server is unwilling to perform the > at Error sending schema update: CN=quota,CN=Schema,CN=Configuration,DC=custombuiltpcs,DC=local The server is unwilling to perform the requested operation
at /usr/share/perl5/EBox/Module/LDAP.pm line 248
EBox::Module::LDAP::_sendSchemaUpdate('EBox::Samba=HASH(0x55d972c89768)', 'Net::LDAP=HASH(0x55d9741201b0)', '/usr/share/zentyal-samba/schema-quota.ldif') called at /usr/share/perl5/EBox/Module/LDAP.pm line 278
EBox::Module::LDAP::_loadSchemasFiles('EBox::Samba=HASH(0x55d972c89768)', 'ARRAY(0x55d9737c2a58)') called at /usr/share/perl5/EBox/Module/LDAP.pm line 267
EBox::Module::LDAP::_loadSchemas('EBox::Samba=HASH(0x55d972c89768)') called at /usr/share/perl5/EBox/Module/LDAP.pm line 341
EBox::Module::LDAP::_performSetup('EBox::Samba=HASH(0x55d972c89768)') called at /usr/share/perl5/EBox/Samba.pm line 669
EBox::Samba::_regenConfig('EBox::Samba=HASH(0x55d972c89768)') called at /usr/share/perl5/EBox/Module/Base.pm line 234
eval {...} at /usr/share/perl5/EBox/Module/Base.pm line 233
EBox::Module::Base::save('EBox::Samba=HASH(0x55d972c89768)') called at /usr/share/perl5/EBox/GlobalImpl.pm line 649
eval {...} at /usr/share/perl5/EBox/GlobalImpl.pm line 648
EBox::GlobalImpl::saveAllModules('EBox::GlobalImpl=HASH(0x55d971d7fa30)', 'progress', 'EBox::ProgressIndicator=HASH(0x55d96dacc7d0)') called at /usr/share/perl5/EBox/Global.pm line 95
EBox::Global::AUTOLOAD('EBox::Global=HASH(0x55d96da9e330)', 'progress', 'EBox::ProgressIndicator=HASH(0x55d96dacc7d0)') called at /usr/share/zentyal/global-action line 32
eval {...} at /usr/share/zentyal/global-action line 30
2024/04/29 16:57:08 ERROR> GlobalImpl.pm:653 EBox::GlobalImpl::saveAllModules - Failed to save changes in module samba: Error sending schema update: CN=quota,CN=Schema,CN=Configuration,DC=custombuiltpcs,DC=loca>2024/04/29 16:57:08 INFO> Base.pm:231 EBox::Module::Base::save - Restarting service for module: logs
2024/04/29 16:57:09 ERROR> GlobalImpl.pm:728 EBox::GlobalImpl::saveAllModules - The following modules failed while saving their changes, their state is unknown: samba at The following modules failed while savi>EBox::GlobalImpl::saveAllModules('EBox::GlobalImpl=HASH(0x55d971d7fa30)', 'progress', 'EBox::ProgressIndicator=HASH(0x55d96dacc7d0)') called at /usr/share/perl5/EBox/Global.pm line 95
EBox::Global::AUTOLOAD('EBox::Global=HASH(0x55d96da9e330)', 'progress', 'EBox::ProgressIndicator=HASH(0x55d96dacc7d0)') called at /usr/share/zentyal/global-action line 32
eval {...} at /usr/share/zentyal/global-action line 30

**Comment by Baltimorepc** [comment]
do i have to add something manually to the main samba server for zentyal to work correctly with it or something

**Comment by djoven89** [comment]
Hi,

We have fixed this issue in [this commit](https://github.com/zentyal/zentyal/commit/cf5c9c140b9d0cf74720b12977d185f3d55f2cba), however, we did not release the package yet because we are working on another bug in this module. So, if someone needs to deploy and configure the Zentyal server as soon as possible, you can just modify the source file `/usr/share/perl5/EBox/Samba/Provision.pm` with the content of the mentioned commit. After that, you can try to enable the module so the join action can be done.

Also, we are writing a documentation that explains the entire process of configuring Zentyal as an Additional Domain Controller.

**Comment by Baltimorepc** [comment]
Thank you. Since i have waited this long and so far i get a new error each time i will just wait for use to release it. if it isnt to much trouble to possibly comment on this thread when its release that would be great. Once again thank you and i appreciate all use do

**Comment by djoven89** [comment]
Hi,

We have released the package `zentyal-samba 8.0.1` that fixes a few bugs, including the ADC join. Also, we have created a new section in the official documentation for how-to. The first doc is about how to configure a Zentyal as an Additional Domain Controller, Below are the link for both languages:

* https://doc.zentyal.org/en/how-to-adc.html
* https://doc.zentyal.org/es/how-to-adc.html

Again, thank you for reporting the issue and for helping us to debug it.

**Comment by Baltimorepc** [comment]
Thank you I will try when I get off 



Sent from Yahoo Mail for iPhone


On Thursday, May 16, 2024, 6:53 AM, Daniel Joven ***@***.***> wrote:




Hi,

We have released the package zentyal-samba 8.0.1 that fixes a few bugs, including the ADC join. Also, we have created a new section in the official documentation for how-to. The first doc is about how to configure a Zentyal as an Additional Domain Controller, Below are the link for both languages:
   
   - https://doc.zentyal.org/en/how-to-adc.html
   - https://doc.zentyal.org/es/how-to-adc.html

Again, thank you for reporting the issue and for helping us to debug it.
